src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberFieldSymbol.cs

```diff
@@ -481,7 +481,8 @@ internal sealed override TypeWithAnnotations GetFieldType(ConsList<FieldSymbol>
                             fieldsBeingBound = new ConsList<FieldSymbol>(this, fieldsBeingBound);
 
                             var initializerBinder = new ImplicitlyTypedFieldBinder(binder, fieldsBeingBound);
-                            var initializerOpt = initializerBinder.BindInferredVariableInitializer(diagnostics, RefKind.None, (EqualsValueClauseSyntax)declarator.Initializer, declarator);
+                            var executableBinder = new ExecutableCodeBinder((EqualsValueClauseSyntax)declarator.Initializer, this, initializerBinder);
```

Comment:
For future compiler team reviewers, this matches non-inferred variable binding: https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Binder/Binder_Initializers.cs,312. This path is missing an ExecutableCodeBinder, which means that things like `SwitchExpressionBinder`s aren't created for the code in the initializer. That then causes the null ref in the bug report.